lib/Dialect/ModArith/IR/ModArithDialect.cpp

```diff
@@ -158,6 +163,168 @@ void ConstantOp::print(OpAsmPrinter &p) {
   p.printType(getOutput().getType());
 }
 
+OpFoldResult ConstantOp::fold(FoldAdaptor adaptor) {
+  // constant(c0) -> c0 mod q
+  APInt cst = adaptor.getValue();
+
+  auto modType = dyn_cast_if_present<ModArithType>(getType());
+  if (!modType) return {};
+
+  // Retrieve the modulus value and its bit width
+  APInt modulus = modType.getModulus().getValue();
+  unsigned modBitWidth = modulus.getBitWidth();
+
+  // Adjust cst's bit width to match modulus if necessary
+  if (cst.getBitWidth() != modBitWidth) {
+    cst = cst.zext(modBitWidth);
+  }
```

Comment:
```suggestion
  cst = cst.zextOrTrunc(modBitWidth);
```

I think this is the right method

Reply:
Similar elsewhere

Reply:
lol